assets/src/domain/eventEditor/ui/datetimes/datesList/cardView/DateCard.tsx

```diff
@@ -1,42 +1,65 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import { __ } from '@wordpress/i18n';
-import { CalendarDateSwitcher } from '@appCalendars/dateDisplay';
 
 import DateActionsMenu from '../actionsMenu/DateActionsMenu';
-import { EntityActionsMenuLayout } from '@appLayout/entityActionsMenu';
-
+import Details from './Details';
+import EntityCard from '@appLayout/EntityCard';
+import { CalendarDateSwitcher, EditDateRangeButton } from '@appCalendars/dateDisplay';
 import { DatetimeProvider } from '@edtrServices/context/DatetimeContext';
+import { EntityActionsMenuLayout } from '@appLayout/entityActionsMenu';
+import { getPropsAreEqual } from '@appServices/utilities';
 import { getStatusTextLabel, statusBgColorClassName } from '@sharedEntities/datetimes/helpers';
-
-import EntityCard from '@appLayout/EntityCard';
 import { useDatesListFilterState } from '@edtrServices/filterState';
-import { getPropsAreEqual } from '@appServices/utilities';
-import Details from './Details';
-import type { DateItemProps } from '../types';
+import { useDatetimeMutator } from '@edtrServices/apollo/mutations';
 import { useMemoStringify } from '@application/services/hooks';
+import { useTimeZoneTime } from '@appServices/hooks';
+import type { DateItemProps } from '../types';
 
 const DateCard: React.FC<DateItemProps> = ({ entity: date }) => {
-	const bgClassName = statusBgColorClassName(date);
 	const { displayStartOrEndDate } = useDatesListFilterState();
+	const { updateEntity } = useDatetimeMutator(date.id);
+	const { siteTimeToUtc } = useTimeZoneTime();
+
+	const onEditHandler = useCallback(
+		(dates: string[]): void => {
+			const [start, end] = dates;
+			// convert start & end dates to proper UTC "startDate" and "endDate"
+			const startDate = siteTimeToUtc(new Date(start)).toISOString();
+			const endDate = siteTimeToUtc(new Date(end)).toISOString();
+			updateEntity({ startDate, endDate });
+		},
+		[date.cacheId]
```

Comment:
Some hook deps are missing

Reply:
thnx
I assume you are talking about `updateEntity` ?
I didn't realize that that could change from render to render.
`siteTimeToUtc` shouldn't ever change since it only relies on site config so that's why I left that out.

Reply:
Yes `updateEntity` can change because of the entity relations.